app/controllers/users/contact_details.js

```diff
@@ -0,0 +1,165 @@
+import Ember from "ember";
+import AjaxPromise from "stock/utils/ajax-promise";
+import AsyncMixin, { ERROR_STRATEGIES } from "stock/mixins/async";
+import TitleAndLanguageMixin from "stock/mixins/grades_option";
+import ImageUploadMixin from "stock/mixins/image_upload";
+
+export default Ember.Controller.extend(
+  AsyncMixin,
+  ImageUploadMixin,
+  TitleAndLanguageMixin,
+  {
+    user: Ember.computed.alias("model.user"),
+    userService: Ember.inject.service(),
+
+    emailIdNotPresent: Ember.computed("user.email", function() {
+      let email = this.get("user.email");
+      return !/^[^@\s]+@[^@\s]+/.test(email);
+    }),
+
+    mobileNotPresent: Ember.computed(
+      "user.mobileWithoutCountryCode",
+      function() {
+        let mobile = this.get("user.mobileWithoutCountryCode");
+        return !/^[456789]\d{7}/.test(mobile);
+      }
+    ),
+
+    districts: Ember.computed(function() {
+      return this.get("store")
+        .peekAll("district")
+        .sortBy("name");
+    }),
+
+    getUpdatedDistrictParams() {
+      let address_attributes = {
+        district_id: this.get("user.associatedDistrict.id") || null
+      };
+      return { address_attributes };
+    },
+
+    saveImage() {
+      let image = this.get("newUploadedImage");
+      if (image) {
+        return image.save();
+      }
+    },
+    deleteOldImage(img) {
```

Comment:
this is just deleting image if image its present or not. 
rename this to `deleteImage(img)`

and the logic of checking weather image is present or not is better to be performed from the place where its called.
with that, this method will have a single responsibility of only deleting the image

See if you can move it to service